cards.js

```diff
@@ -1,23 +1,69 @@
-var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"];
-var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"];
-
-function shuffle(array){
-  var currentIndex = array.length;
-  var randomIndex;
-  var temporaryValue;
-  while(currentIndex > 0){
-    randomIndex   = Math.floor(Math.random() * currentIndex);
-    currentIndex -= 1;
-
-    temporaryValue       = array[currentIndex];
-    array[currentIndex]  = array[randomIndex];
-    array[randomIndex]   = temporaryValue;
+  var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"]; // all the values in the deck
+  var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"]; // all the suits in the deck
+  var deck = []
+
+
+  //this is my array shuffle method
+  function shuffle(array){
+    var currentIndex = array.length;
+    var randomIndex;
+    var temporaryValue;
+    while(currentIndex > 0){
+      randomIndex   = Math.floor(Math.random() * currentIndex);
+      currentIndex -= 1;
+      temporaryValue       = array[currentIndex];
+      array[currentIndex]  = array[randomIndex];
+      array[randomIndex]   = temporaryValue;
+    }
+    return array;
+  }
+
+
+
+  function buildDeck(){
+        for (var i = 0; i < values.length; i++) {       // loop through the length of values and execute the following after each iteration
```

Comment:
per my comment above, you'd want to define `deck` to be a new empty array inside this function.